### ISSUE TITLE
Create 'relaton-ogc' to fetch OGC documents

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/relaton-ogc#1]
@gbuehler mentioned that there is an existing OGC database of all publications.

We can take advantage of that either by:

1. directly querying against a web endpoint that exposes that data source, or
2. that data source can be transformed into a static YAML file like https://github.com/CalConnect/standards.calconnect.org/blob/master/_input/admin.yaml

Either way we will need a new gem.

@gbuehler how can we obtain this database? If it's a static one, how do we keep it updated?

**Comment by gbuehler** [comment]
@ronaldtse The endpoint always gives a JSON dump of the current Standards, Best Practice, Engineering Reports and Discussion Papers. It can be found at:  https://portal.opengeospatial.org/public_ogc/api/docs.php

**Comment by ronaldtse** [comment]
@gbuehler that's great. Does the `docs.php` endpoint also allow querying, such as fetching by document identifier?

**Comment by ronaldtse** [comment]
I was going to say that @ghobona mentioned that the JSON dump can also be found at:
https://github.com/opengeospatial/NamingAuthority/blob/master/incubation/bibliography/bibliography.json

**Comment by ronaldtse** [comment]
And we have a question on the "type" attribute. These values from “type” seem to have some meaning.

AS => Abstract Specification
BP => Best Practice
CAN => ?
CC => ?
CR => ?
CS => Community Standard
DP => Discussion Paper
DP-Draft => Draft DP
IPR => ?
IS => ? 
ISC => ?
ISx => ?
Notes => Notes?
ORM => ?
PC => ?
PER => ?
POL => ?
POL-NTS => ?
Primer => Primer?
Profile => Profile
RFC => (Isn’t RFC a document stage?)
Retired => (Retired is a stage?)
SAP => ?
TS => ?
WhitePaper => White paper?

The D-* ones seem to be Drafts?

D-AS => Draft AS
D-ATB => Draft ?
D-BP => Draft BP
D-DP => Draft 
D-IS => ?
D-ISC => ?
D-ORM => ?
D-Profile => Draft Profile
D-RFC => Draft RFC?
D-RP => ?
D-SAP => ?

**Comment by ronaldtse** [comment]
Thanks @ghobona for the reply. The types are listed here now:

```ruby
{
  "AS" => "Abstract Specification",
  "BP" => "Best Practice",
  "CAN" => "Candidate Standard",
  "CC" => "Conformance Class",
  "CR" => "Change Request",
  "CS" => "Community Standard",
  "DP" => "Discussion Paper",
  "DP-Draft" => "Draft Discussion Paper",
  "IPR" => "Interoperability Program Report - Engineering Specification",
  "IS" => "Implementation Standard",
  "ISC" => "Implementation Standard Corrigendum",
  "ISx" => "Extension Package Standard",
  "Notes" => "Notes",
  "ORM" => "OGC Reference Model",
  "PC" => "Profile Corrigendum",
  "PER" => "Public Engineering Report",
  "POL" => "Policy",
  "POL-NTS" => "Policy - Name Type Specification",
  "Primer" => "Primer",
  "Profile" => "Profile",
  "RFC" => "Request for Comment",
  "Retired" => "Retired document",
  "SAP" => "Standard Application Profile",
  "TS" => "Test Suite",
  "WhitePaper" => "Whitepaper"
}
```

The "D-" prefix means that a document is deprecated.

**Comment by andrew2net** [comment]
@ronaldtse should we update the JSON dump daily as we do in `relaton-nist` https://github.com/relaton/relaton-nist/issues/20#issue-449410651 ?

**Comment by ronaldtse** [comment]
@andrew2net yes we should to ensure it is updated. I believe we can use ETag cache checks to ensure the remote response is updated before fetching the full data again.

**Comment by andrew2net** [comment]
@ronaldtse in the JSON dump there `"identifier"=>"18-073r2"`. Assume we should use `identifier` to search documents. Do we need to fetch references with year included? like `18-073r2:2019`

**Comment by ronaldtse** [comment]
OGC references "18-073r2" implies it is edition 2, so the year is not necessary. Thanks!

**Comment by andrew2net** [comment]
The endpoint https://portal.opengeospatial.org/public_ogc/api/docs.php returns 982 documents. 
The https://github.com/opengeospatial/NamingAuthority/blob/master/incubation/bibliography/bibliography.json returns 979.
The first one returns 3 documents more but doesn't support ETag. Which one should we use?

**Comment by andrew2net** [comment]
> Thanks @ghobona for the reply. The types are listed here now:
> 
> ```ruby
> {
>   "AS" => "Abstract Specification",
>   "BP" => "Best Practice",
>   "CAN" => "Candidate Standard",
>   "CC" => "Conformance Class",
>   "CR" => "Change Request",
>   "CS" => "Community Standard",
>   "DP" => "Discussion Paper",
>   "DP-Draft" => "Draft Discussion Paper",
>   "IPR" => "Interoperability Program Report - Engineering Specification",
>   "IS" => "Implementation Standard",
>   "ISC" => "Implementation Standard Corrigendum",
>   "ISx" => "Extension Package Standard",
>   "Notes" => "Notes",
>   "ORM" => "OGC Reference Model",
>   "PC" => "Profile Corrigendum",
>   "PER" => "Public Engineering Report",
>   "POL" => "Policy",
>   "POL-NTS" => "Policy - Name Type Specification",
>   "Primer" => "Primer",
>   "Profile" => "Profile",
>   "RFC" => "Request for Comment",
>   "Retired" => "Retired document",
>   "SAP" => "Standard Application Profile",
>   "TS" => "Test Suite",
>   "WhitePaper" => "Whitepaper"
> }
> ```
> 
> The "D-" prefix means that a document is deprecated.

@ronaldtse not all of the types match to the [grammar](https://github.com/metanorma/metanorma-model-iso/blob/master/grammars/ogc.rnc)
```
DocumentType =
"standard" | "standard-with-suite" | "abstract-specification" | "community-standard" | "profile" | "best-practice" | "engineering-report" | "discussion-paper" | "reference-model" | "user-guide" | "policy" | "guide" | "amendment" | "technical-corrigendum" | "administrative"
```
Shouldn't we add new types to grammar?

**Comment by andrew2net** [comment]
@ronaldtse there are `D-ATB` and `D-RP` types. They are likely deprecated `ATB` and `RP` documents. But there aren't such types in the type list.

**Comment by ronaldtse** [comment]
> The endpoint https://portal.opengeospatial.org/public_ogc/api/docs.php returns 982 documents.
> The https://github.com/opengeospatial/NamingAuthority/blob/master/incubation/bibliography/bibliography.json returns 979.
> The first one returns 3 documents more but doesn't support ETag. Which one should we use?

I think we just use the second one for now. It gets updated weekly or something.

**Comment by ronaldtse** [comment]
> Shouldn't we add new types to grammar?

Yes you are right! We need to update the grammar (both Relaton and in Metanorma). Ping @opoudjis .

> @ronaldtse there are D-ATB and D-RP types. They are likely deprecated ATB and RP documents. But there aren't such types in the type list.

@ghobona @gbuehler could you help let us know what are the D-ATB and D-RP type documents? Thanks!

**Comment by andrew2net** [comment]
@ronaldtse there are duplicates identifiers:
```
 "16-079",
 "12-128r14",
 "12-128r12",
 "12-176r7",
 "04-094",
 "15-078r6",
 "14-115",
 "14-095",
 "06-042",
 "07-147r2",
 "12-066",
 "09-000",
 "07-036",
 "12-006",
 "09-025r1",
 "07-110r4",
 "03-105r1",
 "04-021r3",
 "07-006r1",
 "06-104r4",
 "11-122r1",
 "06-035r1",
 "05-134",
 "03-006r3",
 "01-009",
 "02-058"
```
l checked `12-128r14` and foud 12 entries. It seems the only difference is `title` though.

**Comment by ghobona** [comment]
For `12-128r14` the duplicate identifiers are from Conformance Classes (CC). These are not standards but rather specification elements. So Conformance Classes should not be considered standards.

For `16-079`, the entries are for different stages of the document. The document was first registered as a Discussion Paper (DP) and then it was developed into an Implementation Standard (IS).

For `12-128r12`, the Conformance Classes and Corrigenda have the same identifier as their source standard.

Hopefully this helps.

**Comment by andrew2net** [comment]
> For `16-079`, the entries are for different stages of the document. The document was first registered as a Discussion Paper (DP) and then it was developed into an Implementation Standard (IS).

Should we search by the stages or just fetch the latest document?

> For `12-128r12`, the Conformance Classes and Corrigenda have the same identifier as their source standard.

Should be Corregeda considered standards?

**Comment by ronaldtse** [comment]
@ghobona Thanks! Some follow up questions:

1. Could you also help explain what D-ATB and D-RP documents are?

2. Are Conformance Classes (CC) citable? If so, is there a way of differentiating them other than using a comparison of title? For example, `12-128r14-cc1` or `12-128r14/cc1`?

3. Corrigenda are usually uniquely citable. For example, ISO uses a syntax like "ISO 1234/Corr 1:2015" to uniquely refer to a document. Should OGC corrigenda be also citable?

**Comment by andrew2net** [comment]
How shoud we treat following documents?
- `14-115` both two entries look similar. The same date and type.
- `14-095` two entries have the same type but different dates. Suppose we should take the latest.
- `03-105r1` three entries with the same date and type
- `04-021r3` two entries with the same date and type
- `01-009` two entries with SAP and SP types and the same date
